MT-CTF/capturetheflag
# ctf_map: Convert to modpack; split map maker into separate mod

This PR performs some refactoring in `ctf_map`, mainly converting it into a modpack, and splitting `map_maker.lua` into a separate `map_maker` mod that depends on the now-renamed `ctf_map_core`. Note that map makers would still have to install the whole modpack, because of the aforementioned dependency.

### Rationale

I'm already working on 3-4 map maker features, and realised that the code has already gotten too messy to add any features while retaining readability. So the primary reason for this refactor is to allow adding more features to map maker, but without having to cram 2000 lines in a single file, which would end up looking like Martian vomit. i.e. this refactor would allow future PRs to add more files to `map_maker`, for much better readability and much better code organisation.

Tested, works.

### Testing

- Run CTF. Observe that everything related to maps work correctly.
- Run MTG with the `ctf_map` modpack enabled. Observe that the `*** CTF_MAP IS IN MAP MAKER MODE ***` is still displayed properly, and that you're able to access the map maker GUI.
